commcare_connect/opportunity/models.py

```diff
@@ -70,3 +70,70 @@ class DeliverForm(models.Model):
 
     def __str__(self):
         return self.name
+
+
+class LearnModule(models.Model):
+    app = models.ForeignKey(
+        CommCareApp,
+        on_delete=models.CASCADE,
+        related_name="learn_module",
+        related_query_name="learn_module",
+    )
+    slug = models.SlugField()
+    name = models.CharField(max_length=255)
+    description = models.CharField(max_length=255)
+    time_estimate = models.IntegerField()
+
+
+class UserModule(models.Model):
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name="user_modules",
+        related_query_name="user_module",
+    )
+    module = models.ForeignKey(
+        LearnModule,
+        on_delete=models.PROTECT,
+        related_name="user_modules",
+        related_query_name="user_module",
+    )
+    opportunity = models.ForeignKey(
+        Opportunity,
+        on_delete=models.PROTECT,
+        related_name="user_modules",
+        related_query_name="user_module",
+    )
+    date_completed = models.DateTimeField(blank=True, null=True)
```

Comment:
This should be non-null. Also thinking that just `date` is sufficient as the name.
```suggestion
    date = models.DateTimeField()
```